src/sources/prometheus/mod.rs

```diff
@@ -0,0 +1,204 @@
+use crate::{topology::config::GlobalOptions, Event};
+use futures::{sync::mpsc, Future, Sink, Stream};
+use http::Uri;
+use hyper;
+use hyper_tls::HttpsConnector;
+use serde::{Deserialize, Serialize};
+use std::time::{Duration, Instant};
+use tokio::timer::Interval;
+
+pub mod parser;
+
+#[derive(Deserialize, Serialize, Clone, Debug)]
+struct PrometheusConfig {
+    host: String,
+    #[serde(default = "default_scrape_interval_secs")]
+    scrape_interval_secs: u64,
+}
+
+pub fn default_scrape_interval_secs() -> u64 {
+    15
+}
+
+#[typetag::serde(name = "prometheus")]
+impl crate::topology::config::SourceConfig for PrometheusConfig {
+    fn build(
+        &self,
+        _name: &str,
+        _globals: &GlobalOptions,
+        out: mpsc::Sender<Event>,
+    ) -> crate::Result<super::Source> {
+        Ok(prometheus(self.clone(), out))
+    }
+
+    fn output_type(&self) -> crate::topology::config::DataType {
+        crate::topology::config::DataType::Metric
+    }
+
+    fn source_type(&self) -> &'static str {
+        "prometheus"
+    }
+}
+
+fn prometheus(config: PrometheusConfig, out: mpsc::Sender<Event>) -> super::Source {
+    let out = out.sink_map_err(|e| error!("error sending metric: {:?}", e));
+
+    let task = Interval::new(
+        Instant::now(),
+        Duration::from_secs(config.scrape_interval_secs),
+    )
+    .map_err(|e| error!("timer error: {:?}", e))
+    .map(move |_| {
+        let uri = format!("{}/metrics", config.host).parse::<Uri>().unwrap();
+        let request = hyper::Request::get(uri).body(hyper::Body::empty()).unwrap();
+
+        let https = HttpsConnector::new(4).expect("TLS initialization failed");
+        let client = hyper::Client::builder().build(https);
```

Comment:
@LucioFranco this is the type of thing it'd be nice to centralize in `SinkContext` or somewhere like that.